### PR TITLE
Improvements .gitignore

### PR DESCRIPTION
**Pull request by LoweredgamesDev**
# Objective

I tried to improve gitignore. I am not a programming expert yet, so I don't know all the junk files. If someone provides me with a list of other files, can you point me to it below? I also tried to sort them logically, hope it works.

> This is also my first PR in the history of this engine. and I'm very happy 😄

### CODE REVIEW

**Comment by LoweredgamesDev** [comment]
😄

File: .gitignore [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
-
-# Generated by "examples/window/screenshot.rs"
-**/screenshot-*.png
-
+# Assets
 assets/**/*.meta
 crates/bevy_asset/imported_assets
 imported_assets
 
+# Example
```
Comment (alice-i-cecile) [inline .gitignore:25]:
```suggestion
# Examples
```

File: .gitignore [hunk 2]
```diff
@@ -1,26 +1,33 @@
+# Rust
```
Comment (alice-i-cecile) [inline .gitignore:1]:
```suggestion
# Rust build artifacts
```

**Comment by alice-i-cecile** [comment]
I quite like the reorganization and comments :) I've left a couple of suggestions to try and improve clarity.

File: .gitignore [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
 /target
 crates/*/target
 **/*.rs.bk
+/benches/target
+/tools/compile_fail_utils/target
+
+# Cargo Rust
```
Comment (BD103) [inline .gitignore:8]:
```suggestion
# Cargo
```

(Nit) The "Rust" bit is implied, so I'd personally go with just "Cargo".

File: .gitignore [hunk 4]
```diff
@@ -1,26 +1,33 @@
...
 Cargo.lock
 .cargo/config
 .cargo/config.toml
+
+# Build files IDE
```
Comment (BD103) [inline .gitignore:13]:
```suggestion
# IDE files
```

(Nit) I'd also go with a more generic "IDE files", since some are used for configuration as well.

**Review by BD103 — APPROVED** [review]
Overall a good organizational improvement! Thanks!

I have 2 nits on comment wording, but you don't need to address them for this to be merged. (And you can refute them, if you have a better idea!)

**Comment by LoweredgamesDev** [comment]
> I quite like the reorganization and comments :) I've left a couple of suggestions to try and improve clarity.

Do you know any other junk files?

**Comment by LoweredgamesDev** [comment]
sorry i made a mistake

**Review by alice-i-cecile — APPROVED** [review]
I haven't run into anything else that needs to be ignored. Thanks for the cleanup!

**Pull request merged**